tequilapi/endpoints/sessions.go

```diff
@@ -48,108 +49,75 @@ func NewSessionsEndpoint(sessionStorage sessionStorage) *sessionsEndpoint {
-//     type: string
-//   - in: query
-//     name: date_to
-//     description: Created date to filter the sessions until this date. Formatted in RFC3339 e.g. 2020-07-01T00:00:00Z.
-//     type: string
-//   - in: query
-//     name: direction
-//     description: Direction to filter the sessions by. Possible values are "Provider", "Consumed".
-//     type: string
-//   - in: query
-//     name: service_type
-//     description: Service type to filter the sessions by.
-//     type: string
-//   - in: query
-//     name: status
-//     description: Status to filter the sessions by. Possible values are "New", "Completed".
-//     type: string
-//   - in: query
-//     name: page
-//     description: Page to filter the sessions by.
-//     type: string
 // responses:
 //   200:
 //     description: List of sessions
 //     schema:
 //       "$ref": "#/definitions/SessionListResponse"
-//   400:
-//     description: Bad request
+//   422:
+//     description: Parameters validation error
 //     schema:
-//       "$ref": "#/definitions/ErrorMessageDTO"
+//       "$ref": "#/definitions/ValidationErrorDTO"
 //   500:
 //     description: Internal server error
 //     schema:
 //       "$ref": "#/definitions/ErrorMessageDTO"
 func (endpoint *sessionsEndpoint) List(resp http.ResponseWriter, request *http.Request, _ httprouter.Params) {
-	query := session.NewQuery()
-
-	dateFrom := time.Now().AddDate(0, 0, -30)
-	if fromStr := request.URL.Query().Get("date_from"); fromStr != "" {
-		var err error
-		if dateFrom, err = time.Parse(time.RFC3339, fromStr); err != nil {
-			utils.SendError(resp, err, http.StatusBadRequest)
-			return
-		}
-	}
-	query.FilterFrom(dateFrom)
-
-	dateTo := time.Now()
-	if toStr := request.URL.Query().Get("date_to"); toStr != "" {
-		var err error
-		if dateTo, err = time.Parse(time.RFC3339, toStr); err != nil {
-			utils.SendError(resp, err, http.StatusBadRequest)
-			return
-		}
+	query, errors := contract.NewSessionListQuery(request)
+	if errors.HasErrors() {
+		utils.SendValidationErrorMessage(resp, errors)
+		return
 	}
-	query.FilterTo(dateTo)
 
-	if direction := request.URL.Query().Get("direction"); direction != "" {
-		query.FilterDirection(direction)
+	filter := session.NewFilter()
+	filter.SetStartedFrom(time.Now().AddDate(0, 0, -30))
+	if query.DateFrom != nil {
+		filter.SetStartedFrom(time.Time(*query.DateFrom).Truncate(24 * time.Hour))
 	}
-	if serviceType := request.URL.Query().Get("service_type"); serviceType != "" {
-		query.FilterServiceType(serviceType)
+	filter.SetStartedTo(time.Now())
+	if query.DateTo != nil {
+		filter.SetStartedTo(time.Time(*query.DateTo).AddDate(0, 0, 1).Truncate(24 * time.Hour))
 	}
-	if status := request.URL.Query().Get("status"); status != "" {
-		query.FilterStatus(status)
+	filter.Direction = query.Direction
+	filter.ServiceType = query.ServiceType
+	filter.Status = query.Status
+
+	pageSize := 50
```

Comment:
Can we have these default values as a const somewhere at the top of the file? It's hard to find it in the middle of the function.

Reply:
I created separate PR for these improvements https://github.com/mysteriumnetwork/node/pull/2695